controllers/kserve_customcacert_controller.go

```diff
@@ -0,0 +1,190 @@
+
+	kserveCustomCertConfigMap := &corev1.ConfigMap{}
+	if err := r.Get(ctx, types.NamespacedName{Name: kserveCustomCACertConfigMapName, Namespace: configmap.Namespace}, kserveCustomCertConfigMap); err != nil {
+		if !apierrs.IsNotFound(err) {
+			return err
+		}
+		log.Info(fmt.Sprintf("Creating KServe custom cert ConfigMap (%s) because it detected the creation of the opendatahub global cert ConfigMap (%s)", kserveCustomCACertConfigMapName, odhGlobalCACertConfigMapName))
+		if err := r.Create(ctx, newCaCertConfigMap); err != nil {
+			log.Error(err, "Failed to create KServe custom cert ConfigMap")
+			return err
+		}
+	} else {
+		log.Info("Checking if the data in KServe custom cert ConfigMap differs from the data in the opendatahub global CA cert ConfigMap")
+		existingOdhCustomCertData := kserveCustomCertConfigMap.Data[constants.KServeCACertFileName]
+		if existingOdhCustomCertData == odhCustomCertData {
+			log.Info(fmt.Sprintf("No updates required for KServe custom cert ConfigMap (%s) as the data matches the opendatahub global cert ConfigMap (%s)", kserveCustomCACertConfigMapName, odhGlobalCACertConfigMapName))
+			return nil
+		}
+		log.Info(fmt.Sprintf("Updating KServe custom cert ConfigMap due to changes in the opendatahub global cert ConfigMap (%s)", odhGlobalCACertConfigMapName))
+		if err := r.Update(ctx, newCaCertConfigMap); err != nil {
+			return err
+		}
+		if err := r.deleteStorageSecret(ctx, configmap.Namespace); err != nil {
+			log.Error(err, "Failed to delete the storage-config secret to update the custom cert")
+			return err
+		}
+		log.V(1).Info("Deleted the storage-config Secret to update the custom cert")
+	}
+
+	return nil
+}
+
+// This section is intended for regenerating StorageSecret using new data.
+func (r *KServeCustomCACertReconciler) deleteStorageSecret(ctx context.Context, namespace string) error {
+	foundStorageSecret := &corev1.Secret{}
+
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      constants.DefaultStorageConfig,
+		Namespace: namespace,
+	}, foundStorageSecret)
+
+	if err == nil {
+		err = r.Delete(ctx, foundStorageSecret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkOpenDataHubGlobalCertCAConfigMapName(objectName string) bool {
+	return objectName == odhGlobalCACertConfigMapName
+}
+
+// reconcileOpenDataHubGlobalCertConfigMap filters out all ConfigMaps that are not the OpenDataHub global certificate ConfigMap.
+func reconcileOpenDataHubGlobalCACertConfigMap() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			objectName := e.Object.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			objectName := e.Object.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
```

Comment:
Could you please explain on which type of event we are trying to act on here. I had never used GenericFunc before.

Reply:
You are right. GenericEvent can include various types of events such as creations, deletions, or updates of Kubernetes API objects. However, it already has those events(create/update/delete) so I think we don't need this.
I removed the event.